youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,158 @@
+    }, {
+        'url': 'https://video.lasentinella.gelocal.it/locale/vestigne-centra-un-auto-e-si-ribalta/138931/139466',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        player_data = re.findall(
+            r'''PlayerFactory\.setParam\('(?P<type>format|param)',\s*'(?P<name>(?:audio|video|image|mp4).*?)',\s*'(?P<val>.+?)'\);''',
+            webpage)
+
+        formats = []
+        thumb = None
+        for t, n, v in player_data:
+            if t == 'format':
+                # http direct formats
+                fmt = re.match(r'(?:video|mp4)(?:-rrtv-)?(\d+)?-?(\d+)?$', n)
+                if fmt:
+                    formats.append({
+                        'format_id': n if 'video' in n else 'video-%s' % n,
+                        'url': v,
+                        'ext': 'mp4',
+                        'protocol': 'https',
+                        'height': int_or_none(fmt.group(1)) or 360,
+                        'tbr': int_or_none(fmt.group(2)) or (
+                            4500 if fmt.group(1) == '1080' else 650),
+                    })
+                    continue
+                # hls formats
+                fmt = re.match(r'(video|audio)-(hls|ipad)-', n)
+                if fmt:
+                    ext = 'mp4' if fmt.group(1) == 'video' else 'm4a'
+                    formats.extend(self._extract_m3u8_formats(
+                        v, video_id, ext, m3u8_id=n, fatal=False))
+                    continue
+                # hds formats
+                if 'video-hds-' in n:
+                    f4m_formats = self._extract_f4m_formats(
+                        '%s?hdcore=3.7.0' % v, video_id, f4m_id=n, fatal=False)
+                    for entry in f4m_formats:
+                        entry.update({'extra_param_to_segment_url': 'hdcore=3.7.0'})
+                    formats.extend(f4m_formats)
+                    continue
+                # mp3 audio
+                if n == 'audio-rrtv-mp3':
+                    formats.append({
+                        'format_id': 'audio-mp3',
+                        'url': v,
+                        'tbr': 128,
+                        'ext': 'mp3',
+                        'vcodec': 'none',
+                        'acodec': 'mp3',
+                    })
+            elif t == 'param' and n in ['image_full', 'image']:
+                thumb = v
+
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': self._html_search_meta('twitter:title', webpage) or self._og_search_title(webpage),
+            'description': self._html_search_meta(
+                ['twitter:description', 'og:description', 'description'],
+                webpage, default=None),
+            'timestamp': parse_iso8601(self._og_search_property(
+                'published_time', webpage, default='').strip()),
```

Comment:
- is it common that videos would not have `og:published_time`?
- do you have an example where the `strip` method call is needed?

Reply:
<meta property="og:published_time" content="2020-10-08T12:25:40 "/>
<meta property="og:modified_time" content="2020-10-08T12:25:40+02:00"/>

as you can see when the timezone is not set (not in every page) there is a space at the end and it makes the iso6601 method fail.

It's in one of the test that I put to only_matching. Should I restore it and comment it explaining the .split() thing?

Reply:
just add a comment above the related `only_matching` test case.

> is it common that videos would not have `og:published_time`?

Reply:
no, I guess it'a always there.

Reply:
then I think it would be better to use `fatal=False` instead of `default=None`.

Reply:
ok

Reply:
looking again at this, there are multiple problems with the `og:published_time`, the timezone is important to calculate the correct timestamp so the `og:published_time` value for embeds is malformed, and also there is a discrepancy between the values from the video page and the values from the embed page, so unless there is a way to determine the correct value, it might be better to drop it.

Reply:
well, if I simply remove the .strip() then if the value is malformed the timestamp is set to None.

Wouldn't that be ok? Set the timestamp only if available and only if not malformad.

Reply:
in the case of https://video.espresso.repubblica.it/tutti-i-video/01-ted-villa/14772/14870
the date extracted from `og:published_time` is `2020-10-15`.
while the one that is shown on the webpage `08 ottobre 2020`(i think this the right date).
so it appears that `og:published_time` is also unreliable similar to `videotitle`.

Reply:
well, even in the player setup script there are two dates:
```
"uploadDate": "2020-10-15T18:08:04+02:00",
"contentUrl": "https://media.gedidigital.it/repubblicatv/file/2020/10/08/.......
```

so I guess I'll remove it at that's it. right?

Reply:
> so I guess I'll remove it at that's it. right?

yes, it would be better to remove it.